modular/gater/object_handler.go

```diff
@@ -497,13 +476,55 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 
 	} // else anonymous users can get public object.
 
+	// do the actual download
+	err = g.downloadObject(w, reqCtx)
+	if err != nil {
+		return
+	}
+}
+
+func (g *GateModular) downloadObject(w http.ResponseWriter, reqCtx *RequestContext) error {
+	var (
+		err                       error
+		params                    *storagetypes.Params
+		bucketInfo                *storagetypes.BucketInfo
+		objectInfo                *storagetypes.ObjectInfo
+		isRange                   bool
+		rangeStart, rangeEnd      int64
+		lowOffset                 int64
+		highOffset                int64
+		pieceInfos                []*downloader.SegmentPieceInfo
+		pieceData                 []byte
+		extraQuota, consumedQuota uint64
+		replyDataSize             int
+		dbUpdateTimeStamp         int64
+	)
+	defer func() {
+		if err != nil {
+			// if the bucket exists extra quota when download object, recoup the quota to user
+			if extraQuota > 0 {
+				quotaUpdateErr := g.baseApp.GfSpClient().RecoupQuota(reqCtx.Context(), bucketInfo.Id.Uint64(), extraQuota, sqldb.TimestampYearMonth(dbUpdateTimeStamp))
+				// no need to return the db error to user
+				if quotaUpdateErr != nil {
+					log.CtxErrorw(reqCtx.Context(), "failed to recoup extra quota to user", "error", err)
+				}
+				log.CtxDebugw(reqCtx.Context(), "//"+
+					"success to recoup extra quota to user", "extra quota:", extraQuota)
+			}
+			w.Header().Del(ContentLengthHeader)
+			w.Header().Del(ContentRangeHeader)
+			w.Header().Del(ContentTypeHeader)
+			w.Header().Del(ContentDispositionHeader)
```

Comment:
ditto for ContentDispositionHeader

Reply:
same here.  If ContentDispositionHeader is not cleared, the error page html will be downloaded as a file rather than as a page.

Those slight differences are introduced because of the refactoring, which put "setting ContentDispositionHeader" and "setting ContentTypeHeader" in advance, so that the real actual download process could be reused in both getObjectHandler and universalEndpointHandler